x-pack/plugins/enterprise_search/public/applications/shared/kibana_breadcrumbs/set_breadcrumbs.tsx

```diff
@@ -17,19 +17,26 @@ import { appSearchBreadcrumbs } from './generate_breadcrumbs';
 
 export type TSetBreadcrumbs = (breadcrumbs: EuiBreadcrumb[]) => void;
 
-interface ISetBreadcrumbsProps {
+interface IBreadcrumbProps {
   text: string;
-  isRoot?: boolean;
+  isRoot?: never;
+}
+interface IRootBreadcrumbProps {
+  isRoot: true;
```

Comment:
Should this be `boolean`?

Reply:
This is me trying to do conditional types in Typescript and not really being sure if I hit it? 😬 I  grabbed the idea from [this stackoverflow q](https://stackoverflow.com/questions/44211804/javascript-conditional-inside-typescript-interface). Basically I'm trying to do:

1. if the `isRoot` prop is present (`true`), the `text` prop should not exist, which looks like:
```jsx
<SetBreadcrumbs isRoot>
```

OR:

2. If the `isRoot` prop is not present, `text` is required, which looks like:
```jsx
<SetBreadcrumbs text="Some page">
```

If there's a better way of doing it than `IFoo | IBar` definitely let me know 🤔